doc/api/cli.md

```diff
@@ -223,6 +209,20 @@ added: v13.1.0
 Enable experimental support for a package using `require` or `import` to load
 itself.
 
+### `--experimental-specifier-resolution=mode`
+<!-- YAML
+added: REPLACEME
+-->
+
+Sets the resolution algorithm for resolving ES module specifiers. Valid options
+are `explicit` and `node`.
```

Comment:
If we're changing the name I wonder if it might make sense to change the name of the extension searching resolution to `--experimental-specifier-resolution=legacy` or `--experimental-specifier-resolution=compat` or something like that?

Reply:
Maybe `--experimental-specifier-resolution=require` to make it obvious what it is emulating?

Reply:
I'm pretty sure there was some distaste towards the implication that the flag was for legacy compatibility - the flag exists to test an alternative that could be a viable way _forward_.

Reply:
Would `--experimental-specifier-resolution=detect` capture the semantics?

Reply:
I'm happy to go with what we have though, and have approved the PR.

Reply:
`require` or `commonjs` or `detect` all would be acceptable to me; `legacy` would not.

Reply:
errr, if you want a new name, why not `implicit` to mirror `explicit`?

Reply:
Everyone cool with me resolving this for now and discussing changing names in a different thread? I'd prefer we don't spend time bikeshedding and instead get this landed and shipped asap